src/SDK/SmartDetectorsSDK/TelemetryDataClient/ILogAnalyticsClient.cs

```diff
@@ -13,32 +13,22 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors
     using System.Threading.Tasks;
 
     /// <summary>
-    /// An interface for running queries against the relevant telemetry database.
+    /// An interface for running queries against the relevant Log Analytics database.
     /// </summary>
-    public interface ITelemetryDataClient
+    public interface ILogAnalyticsClient
     {
         /// <summary>
         /// Gets or sets the query timeout.
         /// </summary>
         TimeSpan Timeout { get; set; }
 
         /// <summary>
-        /// Run a query against the relevant telemetry database.
+        /// Run a query with a specific timespan against the relevant Log Analytics database.
         /// </summary>
         /// <param name="query">The query to run.</param>
+        /// <param name="dataTimeSpan">A time span to use for limiting the query data range.</param>
```

Comment:
>A time span to use for limiting the query data range [](start = 39, length = 52)

does it mean timespan back from now? you could make it more generic and get start time and end time

Reply:
yes - timespan back from now
the LA query API only supports this type of timerange

---
In reply to: [301939950](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/68#discussion_r301939950) [](ancestors = 301939950)